### PR TITLE
ADD: cluster and reverse complement correction for resfinder 

### PR DESCRIPTION
**Pull request by Vedanth-Ramji**
- Resfinder has gene clusters which can't be passed through RGI using 'contig' mode.
- Gene clusters were identified and were manually assigned ARO numbers.
- 40 gene clusters present.
- 9 genes in reverse complement form also present.
- RC genes were all manually curated.
- Delete get_data_path function

### CODE REVIEW

File: argnorm/data/cluster_rc_correction/notes.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Gene Clusters
+
+- Resfinder has gene clusters (nucleotide sequence with multiple CDSs present) which can't be passed through RGI using 'contig' mode.
+- Gene clusters were identified and were manually assigned ARO numbers.
+- A seperate file with manual curation for gene clusters and RCs was created, and their AROs were updated after concatenating RGI results and genes not in RGI results.
```
Comment (luispedro) [inline argnorm/data/cluster_rc_correction/notes.md:5]: s/seperate/separate/

**Review by luispedro — COMMENTED** [review]
Thanks.

It seems, however, that this could have been done without special-casing `resfinder`? Why is this different from the other manual curation files?

**Comment by luispedro** [comment]
The code for actually running rgi correctly for resfinder is still missing as well.

**Comment by Vedanth-Ramji** [comment]
I just added the code for using the AA version of resfinder.

> It seems, however, that this could have been done without special-casing `resfinder`? Why is this different from the other manual curation files?

The other manual curation files are adding genes that are not present in the mapping tables (i.e. RGI can't map them automatically). The new 'correction' files are changing mappings that are already present in the mapping tables (i.e. correcting RGI as it is not detecting a gene cluster or reverse complement properly and mapping to a wrong ARO number).

File: db_harmonisation/crude_db_harmonisation.py [hunk 2]
```diff
@@ -70,6 +69,16 @@ def fix_ncbi(ncbi_amr_faa):
 
     return ofile
 
+@TaskGenerator
+def fna_to_faa(ifile):
+    ofile = f'./dbs/resfinder.faa'
```
Comment (luispedro) [inline db_harmonisation/crude_db_harmonisation.py:74]: This should not be hardcoded, but

```python
ofile = ifile.replace('.fna', '.faa')
```

**Comment by luispedro** [comment]
> The other manual curation files are adding genes that are not present in the mapping tables (i.e. RGI can't map them automatically). The new 'correction' files are changing mappings that are already present in the mapping tables (i.e. correcting RGI as it is not detecting a gene cluster or reverse complement properly and mapping to a wrong ARO number).

I know this, but I don't see why it matters for my question. Why does the code need to special-case `resfinder`? Isn't this just a different way in which the files were generated, but the same code can handle both.

**Review by luispedro — CHANGES_REQUESTED** [review]

File: argnorm/lib.py [hunk 3]
```diff
@@ -16,27 +16,33 @@ def is_number(num):
...
 def get_aro_mapping_table(database):
-    df = pd.read_csv(get_data_path(f'{database}_ARO_mapping.tsv', False), sep='\t')
+    df = pd.read_csv(os.path.join(_ROOT, 'data', f'{database}_ARO_mapping.tsv'), sep='\t')
 
-    manual_curation = pd.read_csv(get_data_path(f'{database}_curation.tsv', True), sep='\t')
+    manual_curation = pd.read_csv(os.path.join(_ROOT, 'data/manual_curation', f'{database}_curation.tsv'), sep='\t')
     manual_curation['Database'] = df['Database']
+    aro_mapping_table = df
+
+    if database != 'resfinder':
```
Comment (luispedro) [inline argnorm/lib.py:26]: Can we remove this `if` statement?

File: argnorm/lib.py [hunk 4]
```diff
@@ -16,27 +16,33 @@ def is_number(num):
...
-    return os.path.join(_ROOT, 'data', path)
-
 def get_aro_mapping_table(database):
-    df = pd.read_csv(get_data_path(f'{database}_ARO_mapping.tsv', False), sep='\t')
+    df = pd.read_csv(os.path.join(_ROOT, 'data', f'{database}_ARO_mapping.tsv'), sep='\t')
 
-    manual_curation = pd.read_csv(get_data_path(f'{database}_curation.tsv', True), sep='\t')
+    manual_curation = pd.read_csv(os.path.join(_ROOT, 'data/manual_curation', f'{database}_curation.tsv'), sep='\t')
     manual_curation['Database'] = df['Database']
+    aro_mapping_table = df
```
Comment (luispedro) [inline argnorm/lib.py:24]: We can avoid this by not having an intermediate object called `df`

**Review by luispedro — CHANGES_REQUESTED** [review]
I still want to remove the special casing of `resfinder`

**Pull request merged**